src/metalookup/core/metadata_manager.py

```diff
@@ -137,12 +110,34 @@ async def run_extractor(extractor: Extractor) -> Union[MetadataTags, Error]:
 
         self.logger.debug("Dispatching extractor calls.")
 
-        # The call to gather (with return_exceptions=False) fails if any of the submitted tasks raise an exception.
-        # This is what we want, as we have to respond with an all or nothing result. I.e. we have metadata extracted
-        # for all fields or none of them.
-        results: tuple[Union[MetadataTags, Error], ...] = await asyncio.gather(
-            *[run_extractor(extractor=extractor) for extractor in self.extractors]
-        )
-        self.logger.debug("Received all extractor results.")
+        try:
+            # The call to gather (with return_exceptions=False) fails if any of the submitted tasks raise an exception.
+            # This is what we want, as we have to respond with an all or nothing result. I.e. we have metadata extracted
+            # for all fields or none of them.
+            results: tuple[Union[MetadataTags, Error], ...] = await asyncio.gather(
+                *[run_extractor(extractor=extractor) for extractor in self.extractors]
+            )
+            self.logger.debug("Received all extractor results.")
+
+            # Make sure we didn't extract meta information from a 404 placeholder site or similar.
+            # This happens after the calls to extractors to prevent awaiting splash and then awaiting the
```

Comment:
Mhh, yeah ill try to clarify that: 

If we do this 404 check before submitting the extractor tasks, then we essentially have the same sequential order as before: The MetadataManager awaits on the splash response, and only if that is not a 404 will it start the extractors (including lighthouse).

To avoid that, we let the extractors run and analyze, but then essentially discard their results afterwards - which admittedly is also not so nice:-(

Maybe it would be better to raise the 502 Exception from within `await Content.har()` or `await Content.html()` then  all extractors that make use of `har` or `html` (which would be almost all of them) would fail individually.

Any opinion?

Reply:
Well, either way, all extractors using splash output would fail. Thus: fail fast. I like the last idea of 502. If one fails, why not pull everyone down into the abyss? :)

Reply:
I created https://github.com/openeduhub/metalookup/issues/155 for this. Eventually that can be resolved together with https://github.com/openeduhub/metalookup/issues/85